tokens/src/commands.rs

```diff
@@ -1397,42 +1484,251 @@ mod tests {
     }
 
     #[test]
-    #[should_panic]
     fn test_read_allocations_malformed() {
-        let pubkey0 = solana_sdk::pubkey::new_rand();
-        let pubkey1 = solana_sdk::pubkey::new_rand();
+        let pubkey0 = pubkey::new_rand();
+        let pubkey1 = pubkey::new_rand();
+
+        // Empty file.
         let file = NamedTempFile::new().unwrap();
+        let mut wtr = csv::WriterBuilder::new().from_writer(&file);
+        wtr.flush().unwrap();
         let input_csv = file.path().to_str().unwrap().to_string();
-        let mut wtr = csv::WriterBuilder::new().from_writer(file);
+        let got = read_allocations(&input_csv, None, false, false);
+        match got {
+            Err(Error::CsvIsEmptyError) => (),
+            _ => unreachable!("expected CsvIsEmptyError, got: {:?}", got),
+        };
+
+        // Missing 2nd column.
+        let file = NamedTempFile::new().unwrap();
+        let mut wtr = csv::WriterBuilder::new().from_writer(&file);
+        wtr.serialize("recipient".to_string()).unwrap();
+        wtr.serialize(pubkey0.to_string()).unwrap();
+        wtr.serialize(pubkey1.to_string()).unwrap();
+        wtr.flush().unwrap();
+        let input_csv = file.path().to_str().unwrap().to_string();
+        let got = read_allocations(&input_csv, None, false, false);
+        match got {
+            Err(Error::CsvError(..)) => (),
+            _ => unreachable!("expected CsvError, got: {:?}", got),
+        };
```

Comment:
A nice way to write this is:
```suggestion
        let got = read_allocations(&input_csv, None, false, false);
        assert!(matches!(got, Err(Error::CsvError(..))));
```

or
```suggestion
        let err_result = read_allocations(&input_csv, None, false, false).unwrap_err();
        assert!(matches!(err_result, Error::CsvError(..)));
```

Reply:
Looks like you can just use the additional `if` syntax to also check the String: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ff785bc2a4a33a51d3978ade39a37bcc

Fwiw, I have a slight preference for using `unwrap_err()` and leaving the Result variant out of the `matches` comparison.

Reply:
I see, applied this to the rest of error-handling (where appropriate), thanks!